### ISSUE TITLE
`@layer` rules that do not have a block and do not have a trailing semi-colon break sourcemaps

### ISSUE THREAD

**Issue by romainmenke** [postcss/postcss#1802]
```css
@layer extensions {
  @layer one, two
}
```

This is equivalent to:

```css
@layer extensions.one, extensions.two;
```

It parses correctly but fails to write a sourcemap :

```
Invalid mapping: {"generated":{"line":3,"column":-1},"source":"a.css","original":{"line":2,"column":16},"name":null}
```

The error goes away when I add a trailing semi-colon :

```css
@layer extensions {
  @layer one, two;
}
```

I don't have any knowledge about sourcemap internals and failed to pinpoint where it is going wrong.

I did add some tests to reproduce this : https://github.com/postcss/postcss/pull/1801

I don't think the parsing the test is needed but I wanted to see how it was parsed.
Maybe I am overlooking something there.

**Comment by ai** [comment]
Thanks for the issue with a lot of details.

I will fix it. But very likely that I will do it after my vacation week later.

**Comment by romainmenke** [comment]
This is not blocking for us, so take your time :)

**Comment by ai** [comment]
Fixed 20122e50

**Comment by romainmenke** [comment]
Thank you @ai 🙇

**Comment by ai** [comment]
The fix was released at 8.4.20.
